tower-balance/src/load/track.rs

```diff
@@ -0,0 +1,77 @@
+use futures::{Future, Poll};
+use std::marker::PhantomData;
+
+/// Attaches `T`-typed trackers to `V` typed values.
+///
+/// This utility allows load metrics to have a protocol-agnostic means to track streams
+/// past their initial response future. For example, if `V` represents an HTTP response
+/// type, an implementaton could add `T`-typed trackers to the HTTP response extensions.
+///
+/// Trackers are intended to be RAII guards that primarily implement `Drop` and update
+/// load metric state as they are dropped.
+///
+/// A base `impl<T, V> Track<T, V> for ()` is provided that drops the tracker
+/// immediately.
+pub trait Track<T, V> {
+    type Tracked;
+
+    /// Attaches a `T`-typed trakcer to a `V`-typed value.
+    fn track(tracker: T, value: V) -> Self::Tracked;
+
+    /// Wraps an `F`-typred Future so that a `T`-typed tracker is attached to its result.
+    fn track_future<F>(tracker: T, future: F) -> TrackFuture<F, Self, T>
```

Comment:
Does this need to be a fn on the trait? Since this is a utility trait, the answer would be yes if `track_future` is ever overridden. However, I don't think it can be overridden in a meaningful way.

Reply:
You're correct. This is provided as a convenience to make construction slightly more ergonomic.  It can probably be changed to `TrackFuture::new`